messaging/src/main/java/org/axonframework/messaging/annotation/MessageHandlerInterceptorDefinition.java

```diff
@@ -89,6 +92,24 @@ public Object handle(Message<?> message, T target) throws Exception {
                     throw e;
                 });
             }
+            finally {
+                if (result instanceof CompletableFuture){
+                    // Invoke the exception handler in case of an exception
+                    ((CompletableFuture<?>) result).exceptionally(exception -> {
+                        try {
+                            ResultParameterResolverFactory.callWithResult(exception, () -> {
+                                if (super.canHandle(message)){
+                                    return super.handle(message, target);
+                                }
+                                return null;
+                            });
+                        } catch (Exception ignored) {
+                         // exception is expected here while handling an exception
```

Comment:
If we want to propagate exceptions from exception handlers, then it should be done here and not ignored. Exceptions from regular, non CF-result handlers are propagated.